ERCS/erc-7564.md

```diff
@@ -0,0 +1,209 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-11-21
+requires: 165
+---
+
+## Abstract
+
+This proposal introduces a smart contract wallet-based approach for managing nfts, focusing on utilizing the programmable features of smart contract wallets for nft asset management. Additionally, it introduces functions such as `nftApprove`, `nftSetApprovalForOneAll`, `nftSetApprovalForAllAll`, `nftGetApproved`, `nftIsApprovedForOneAll`, `nftIsApprovedForAllAll` and `nftTransfer`, which provide enhanced control over nft transactions. This approach seeks to enhance nft management by utilizing the built-in features of smart contract wallets, thus offering a more adaptable, secure, and efficient method for managing token transactions.
+
+
+## Motivation
+
+An externally-owned account (EOA) wallet has no state and code storage, while the smart contract wallet does.
+
+Account abstraction (AA) is a direction of the smart contract wallet, which works around abstract accounts. This ERC can also be an extension based on [ERC-4337](./eip-4337) or as a plug-in for wallets.
+
+The smart contract wallet allows the user's own account to have state and code, bringing programmability to the wallet. We think there are more directions to expand. For example, nft asset management, functional expansion of nft transactions, etc.
+
+The smart contract wallet interface of this ERC is for nft asset management and nft asset approval. It supports the simplenft <!-- TODO --> ERC-X, and [ERC-721](./eip-721) is backward compatible with <!-- TODO --> ERC-X, so it can be compatible with the management of all nfts in the existing market.
+
+The proposal aims to achieve the following goals:
+
+1. NFT assets are allocated and managed by the wallet itself, such as approve function, which are configured by the user’s contract wallet, rather than controlled by the nft asset contract, to avoid some existing ERC-721 contract risks.
+2. Add the `nftTransfer` function, the transaction initiated by the non-smart wallet itself.
+3. Add `nftApprove`, `nftSetApprovalForOneAll`, `nftSetApprovalForAllAll`, `nftGetApproved`, `nftIsApprovedForOneAll`, `nftIsApprovedForAllAll` functions. The user wallet itself supports approve and provides approve. for One nft, all nft of one nft smart contract, all nft assets.
+4. User wallet can choose batch approve and batch transfer.
+5. Users can choose to add hook function before and after their `nftTransfer` to increase the user's more playability.
+6. The user can choose to implement the `nftReceive` function.
+
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+**Compliant contract must implement the [ERC-165](./eip-165) interfaces**
+```solidity
+// SPDX-License-Identifier: GPL-3.0
```

Comment:
We cannot permit GPL'd code in the EIPs repository. Code inline in the document itself _must_ be CC0-1.0. I permit a few more licenses (`MIT`, `Apache-2.0`, etc.) in the assets directory, but again, CC0-1.0 is preferred.

Reply:
thanks so much, I deleted it.
also, Happy Spring Festival